doc/nrf/protocols/matter/getting_started/memory_profiling.rst

```diff
@@ -0,0 +1,371 @@
+.. _ug_matter_device_memory_profiling:
+
+Profiling memory in Matter applications
+#######################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This guide offers useful commands and Kconfig options for configuring memory sizes for stacks, heaps, and Zephyr settings.
+You can learn to measure memory usage and adjust these values according to your project's requirements.
+
+Measuring memory usage
+======================
+
+You can use several predefined Kconfig options and UART shell commands to obtain the current memory statistics from the device.
```

Comment:
Isn't "predefined" here and there a little bit redundant, as all Kconfigs are sort of predefined?